packages/webpack-config-terra/webpack.config.js

```diff
@@ -0,0 +1,250 @@
+const ThemePlugin = require('./lib/postcss/ThemePlugin');
+const getThemeConfig = require('./lib/utils/_getThemeConfig');
+
+/**
+ * Get the locales to be defined by the webpack build, also aggregates translations, but we should remove that in the future.
+ * @param {object} env the env object
+ * @param {string} rootPath root path of the package.
+ */
+const getLocales = ({ disableAggregateTranslations }, rootPath) => {
+  if (!disableAggregateTranslations) {
+    return aggregateTranslations({ baseDir: rootPath });
+  }
+  return undefined;
+};
+
+/**
+ * Get an array of modules that need to be resolved to during the webpack build. Should be removed after aggregate translations is removed.
+ * @param {object} env the env object
+ */
+const getResolveModules = ({ disableAggregateTranslations }) => (
+  [
+    'node_modules',
+    ...disableAggregateTranslations ? [] : ['aggregated-translations'],
+  ]
+);
+
+/**
+ * Determines the correct theme config to usee of the three options. Precedence is. ENV, options config, terra-them.config.js
+ * @param {object} options object containing the override config if provided.
+ */
+const determineThemeConfig = ({ themeConfig: overrideThemeConfig }) => {
+  const defaultTheme = process.env.THEME; // Flexes root theme for theme visual regression testing.
+
+  if (defaultTheme) {
+    return { theme: defaultTheme };
+  }
+  if (overrideThemeConfig) {
+    return overrideThemeConfig;
+  }
+
+  return getThemeConfig();
+};
+
+/**
+ * Provides the webpack dev server options to disable hot reloading. I'm going to be honest, it doesn't alway work.
+ * @param {object} env the env object containing the request to disable hot reloading.
+ */
+const getWebpackDevServerStaticOptions = ({ disableHotReloading }) => (
+  {
+    ...disableHotReloading && {
+      hot: false,
+      inline: false,
+    },
+  }
+);
+
+/**
+ * The extendable webpack config for terra.
+ * @param {object} env environment variables pass to webpack via --env.thing.
+ * @param {object} argv webpack options from the command line arguments, no custom options here.
+ * @param {object} options custom webpack options provided via a config override.
+ */
+const defaultWebpackConfig = (env = {}, argv = {}, options = {}) => {
+  const processPath = process.cwd();
+  /* Get the root path of a mono-repo process call */
+  const rootPath = processPath.includes('packages') ? processPath.split('packages')[0] : processPath;
+
+  const themeConfig = determineThemeConfig(options);
+
+  const production = argv.p;
+  const fileNameStategy = production ? '[name]-[chunkhash]' : '[name]';
+  const chunkFilename = argv['output-chunk-filename'] || fileNameStategy;
+  const filename = argv['output-filename'] || fileNameStategy;
+  const outputPath = argv['output-path'] || path.join(rootPath, 'build');
+  const publicPath = argv['output-public-path'] || '';
+
+  const devConfig = {
+    mode: 'development',
+    entry: {
+      'core-js': 'core-js/stable',
```

Comment:
Do we need to add all of these? I know @ryanthemanuel has been looking at picking the exact polyfills we need to improve performance in IE 11 in IE 10 compatibility mode browsers

Reply:
I removed RAF based on @ryanthemanuel's investigation. If theres more to remove i can do that.

Reply:
Some of what I'm doing is specific to MPages (they have their own set of polyfills they add that we're going to be using instead), so I think we'd need to be more abstract than that unfortunately.  One thing we could look into is running core-js/stable through the babel transpiler.  They have a method to transpile that to just the list of components you need from what your browserslist specifies:

```
    ['@babel/preset-env', { useBuiltIns: 'entry', corejs: { version: 3.1, proposals: true } }],
```

Reply:
I added this in a way that consumers can opt in by adding the config you list here. Basically I changed the entrypoint file to point to a file containing only 'import 'core-js/stable' and updated our babel loader to always load that file even if it is in node modules.

This way we can adhere to the browserlist specified by the consumer and support removing IE support in the future.

https://github.com/cerner/terra-toolkit/pull/444/commits/150ef0fef2dad1b003d66165f966d83ebe1a95ad